modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaUserLocalServiceUtilCheck.java

```diff
@@ -0,0 +1,210 @@
+
+		while (addUserMatcher.find()) {
+			String methodCall = addUserMatcher.group();
+
+			List<String> parameterList = JavaSourceUtil.getParameterList(
+				methodCall);
+
+			if (!_checkMethodCall(content, fileContent, methodCall) ||
+				!((parameterList.size() == 26) ||
+				  (parameterList.size() == 31))) {
+
+				continue;
+			}
+
+			String line = getLine(
+				content, getLineNumber(content, addUserMatcher.start()));
+			String variableName = getVariableName(methodCall);
+
+			content = StringUtil.replace(
+				content, methodCall,
+				_removeParameters(
+					SourceUtil.getIndent(line), methodCall, parameterList,
+					variableName));
+		}
+
+		return content;
+	}
+
+	private boolean _checkMethodCall(
+		String content, String fileContent, String methodCall) {
+
+		String variableName = getVariableName(methodCall);
+
+		if (variableName.equals("UserLocalServiceUtil") ||
+			variableName.equals("UserServiceUtil") ||
+			hasClassOrVariableName(
+				"UserLocalService", content, fileContent, methodCall) ||
+			hasClassOrVariableName(
+				"UserService", content, fileContent, methodCall)) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private String _checkUpdateStatus(
+		String content, String fileContent, String fileName) {
+
+		Matcher updateStatusMatcher = _updateStatusPattern.matcher(content);
+
+		String newContent = content;
+
+		while (updateStatusMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, updateStatusMatcher.start());
+
+			List<String> parameterList = JavaSourceUtil.getParameterList(
+				methodCall);
+
+			if (!_checkMethodCall(
+					content, fileContent, updateStatusMatcher.group(2)) ||
+				(parameterList.size() != 2) ||
+				!_validateParameters(
+					fileName, content, parameterList,
+					new String[] {"long", "int"})) {
+
+				continue;
+			}
+
+			String newMethodCall = JavaSourceUtil.addMethodNewParameters(
+				JavaSourceUtil.getIndent(methodCall), new int[] {2},
+				updateStatusMatcher.group(),
+				new String[] {"ServiceContextThreadLocal.getServiceContext()"},
+				parameterList);
+
+			newContent = StringUtil.replace(
+				newContent, methodCall, newMethodCall);
+		}
+
+		return newContent;
+	}
+
+	private String _removeParameters(
+		String indent, String methodCall, List<String> parameterList,
+		String variableName) {
+
+		parameterList.remove(7);
+		parameterList.remove(7);
+
+		StringBundler sb = new StringBundler();
+
+		sb.append(variableName);
+		sb.append(".addUser(\n");
+		sb.append(indent);
+		sb.append(StringPool.TAB);
+		sb.append(parameterList.get(0));
+
+		for (int i = 1; i < parameterList.size(); i++) {
+			if ((i % 4) != 0) {
+				sb.append(StringPool.COMMA_AND_SPACE);
+				sb.append(parameterList.get(i));
+
+				continue;
+			}
+
+			sb.append(StringPool.COMMA);
+			sb.append(StringPool.NEW_LINE);
+			sb.append(indent);
+			sb.append(StringPool.TAB);
+			sb.append(parameterList.get(i));
+		}
```

Comment:
@michaelmdelcavalcanti , please inver the logic to be more understandable what you are doing with this.

Reply:
Done.